app/src/main/java/org/mozilla/fenix/components/Components.kt

```diff
@@ -142,10 +142,9 @@ class Components(private val context: Context) {
         AddonManager(core.store, core.engine, addonCollectionProvider, addonUpdater)
     }
 
-    val sitePermissionsStorage by lazyMonitored {
-        SitePermissionsStorage(context, context.components.core.engine)
+    val geckoSitePermissionsStorage by lazyMonitored {
+        GeckoSitePermissionsStorage(core.geckoRuntime, OnDiskSitePermissionsStorage(context))
```

Comment:
nit: Would prefer if we moved all things `Gecko` to Core.kt where the runtime and engine are created as this depends on it directly.

Reply:
Addressed!